### PR TITLE
Convert elasticsearch-rest-6.4 Test from groovy tp java.

### PR DESCRIPTION
**Pull request by huange7**
Convert elasticsearch-rest-6.4 Test from groovy tp java.

### CODE REVIEW

File: instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  static ElasticsearchContainer elasticsearch;
+
+  static HttpHost httpHost;
+
+  static RestClient client;
+
+  static ObjectMapper objectMapper;
+
+  @BeforeAll
+  static void setupSpec() {
```
Comment (laurit) [inline instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java:41]: `setupSpec` and `cleanupSpec` are spock method, for java tests it is better to use `setUp` and `tearDown` or `cleanUp`

File: instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    Exception[] exception = {null};
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    ResponseListener responseListener =
+        new ResponseListener() {
+          @Override
+          public void onSuccess(Response response) {
+            testing.runWithSpan(
+                "callback",
+                () -> {
+                  requestResponse[0] = response;
```
Comment (laurit) [inline instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java:122]: We create the `callback` span here to verify that context is propagated into the `onSuccess` method. You can move the `requestResponse[0] = response;` out of the `runWithSpan`, then you don't need to do the array trick and can use `requestResponse = response;`.

File: instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                  countDownLatch.countDown();
+                });
+          }
+
+          @Override
+          public void onFailure(Exception e) {
+            testing.runWithSpan(
+                "callback",
+                () -> {
+                  exception[0] = e;
```
Comment (laurit) [inline instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java:132]: same as with requestResponse

File: instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java (hunk 2 above)
Comment (huange7) [reply instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java:122]: Even if I move it to the `onSuccess` method, it still seems to need to use the array.

Comment (laurit) [reply instrumentation/elasticsearch/elasticsearch-rest-6.4/javaagent/src/test/java/ElasticsearchRest6Test.java:122]: sorry my bad

**Review by laurit — APPROVED** [review]

**Comment by huange7** [comment]
why checks always in progress, is it normal?

**Comment by huange7** [comment]
> why checks always in progress, is it normal?

all done.  ignore my comment

**Comment by laurit** [comment]
> > why checks always in progress, is it normal?
> 
> all done. ignore my comment

@huange7 it was stuck downloading gradle caches, I canceled the build and reran the stuck jobs.

**Comment by huange7** [comment]
> > > why checks always in progress, is it normal?
> > 
> > 
> > all done. ignore my comment
> 
> @huange7 it was stuck downloading gradle caches, I canceled the build and reran the stuck jobs.

@laurit Thanks a lot.

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by mateuszrzeszutek** [comment]
Thanks @huange7 !

**Pull request merged**